### ISSUE TITLE
Database access AWS RDS Proxy MS SQL Server support

### ISSUE THREAD

**Issue by greedy52** [gravitational/teleport#17339]
What would you like Teleport to do?
[Amazon RDS Proxy now supports Amazon RDS for SQL Server](https://aws.amazon.com/about-aws/whats-new/2022/09/amazon-rds-proxy-rds-sql-server/
)

What problem does this solve?
RDS Proxy general support is implemented in #9824. This issue is used to track the RDS Proxy support for MS SQL Server which requires engine (and maybe 3rd party driver) support for AWS RDS IAM auth.

**Comment by r0mant** [comment]
@greedy52 With RDS Proxy for SQL Server, does that remove requirement for having to use Kerberos auth with the database? Would Teleport just need to authenticate with the RDS Proxy using IAM token, which would then in turn authenticate with SQL Server itself?

**Comment by greedy52** [comment]
> @greedy52 With RDS Proxy for SQL Server, does that remove requirement for having to use Kerberos auth with the database? Would Teleport just need to authenticate with the RDS Proxy using IAM token, which would then in turn authenticate with RDS SQL Server itself?

Yes, that's correct. Teleport can use IAM Auth to connect RDS Proxy. RDS Proxy internally uses saved user/password (saved in Secerts Manager secerts) to connect to SQL Server. I was a little excited about this that we can test out SQL Server (at least client side flows) without Kerberos auth, if we can get it working. 

Though, a special `accessToken` attribute is used when passing the IAM Auth token to the SQL Server RDS Proxy node. The AWS doc mentions JDBC, ODBC, and .net client has this attribute. We have to see if our go driver works.
